nengo_viz/swi.py

```diff
@@ -402,14 +408,18 @@ def start(cls, port=80, asynch=True, addr='', browser=False):
             # shut down any remaining threads
             if asynch and server.requests is not None:
                 for _, sock in server.requests:
```

Comment:
Yeah, I was wondering about list copying being atomic, but I couldn't find any usable information on it before I decided that writing the implementation with the lock will quicker than figuring out the answer.